core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -703,7 +720,67 @@ public void setDefaultTTL(int defaultTTL) {
+     */
+    public void verifyAndCreateTempTables() {
+        ZonedDateTime currentBlock = ZonedDateTime.ofInstant(Instant.ofEpochMilli(DateTimeService.now.get().getMillis()), UTC)
+                .with(DateTimeService.startOfPreviousEvenHour());
+
+        ZonedDateTime lastStartupBlock = currentBlock.plus(6, ChronoUnit.HOURS);
+        verifyAndCreateTempTables(currentBlock, lastStartupBlock).await();
+    }
+
+    @Override
+    public Completable verifyAndCreateTempTables(ZonedDateTime startTime, ZonedDateTime endTime) {
+        Set<Long> timestamps = new HashSet<>();
+
+        while(startTime.isBefore(endTime)) {
+            // Table sizes are not configurable at this point
+            timestamps.add(startTime.toInstant().toEpochMilli());
+            startTime = startTime.plus(2, ChronoUnit.HOURS);
+        }
+
+        return Completable.fromObservable(dataAccess.createTempTablesIfNotExists(timestamps));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Completable compressBlock(long startTimeSlice, int pageSize, int maxConcurrency) {
+        return Completable.fromObservable(
+                dataAccess.findAllDataFromBucket(startTimeSlice, pageSize, maxConcurrency)
+                        .flatMap(rows -> rows
+                                .publish(p -> p.window(
+                                        p.map(Row::getPartitionKeyToken)
+                                                .distinctUntilChanged()))
+                                .concatMap(o -> {
+                                    Observable<Row> sharedRows = o.share();
+                                    Observable<CompressedPointContainer> compressed =
+                                            sharedRows.compose(new TempTableCompressTransformer(startTimeSlice));
+                                    Observable<Row> keyTake = sharedRows.take(1);
+
+                                    return compressed.zipWith(keyTake, (cpc, r) -> {
+                                        MetricId<?> metricId =
+                                                new MetricId(r.getString(0), MetricType.fromCode(r.getByte(1)),
+                                                        r.getString(2));
+                                        return dataAccess
+                                                .insertCompressedData(metricId, startTimeSlice, cpc, getTTL(metricId))
+                                                .mergeWith(updateMetricExpiration(metricId).map(rs -> null));
+                                    });
+                                }), maxConcurrency)
+                        .flatMap(rs -> rs)
+                        .doOnError(Throwable::printStackTrace)
+                        .doOnCompleted(() -> {
+                            log.infof("Compress part completed");
+                            dataAccess.dropTempTable(startTimeSlice)
```

Comment:
I noticed that ```dropTempTable``` has no retry logic. We need to make sure these tables get deleted. What happens if the query fails?